### PR TITLE
[PM-4760] Admin Recovery Permissions prompted to set MP.

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

If elevating user’s role to Custom role with the “Manage Account Recovery” permission, the user needs to be prompted to set a master password. 
This check is made on Sync.

## Code changes

Add force password reset check on sync complete. This check also runs on TabsPage OnAppearing, we will leverage existing broadcast subscription to "synccomplete" to also check for password reset. 

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/App/Pages/TabsPage.cs [hunk 1]
```diff
@@ -94,6 +94,7 @@ protected override async void OnAppearing()
                 if (message.Command == "syncCompleted")
                 {
                     Device.BeginInvokeOnMainThread(async () => await UpdateVaultButtonTitleAsync());
+                    await ForcePasswordResetIfNeededAsync();
```
Comment (fedemkr) [inline src/App/Pages/TabsPage.cs:97]: ⚠️ Keep in mind that given that you're in an action in the broadcaster subscription, this act as an `async void` so if `ForcePasswordResetIfNeededAsync()` throws an exception it'll crash the application. I don't know if that's what you expect, but I'd think that we should wrap this into a try...catch and log the exception and then show an alert to the user that an error happened and maybe log the user out, IDK which would be the appropriate flow if this method throws.

**Review by fedemkr — CHANGES_REQUESTED** [review]

File: src/App/Pages/TabsPage.cs (hunk 1 above)
Comment (andrebispo5) [reply src/App/Pages/TabsPage.cs]: added a exception log if that ever occurred.

File: src/App/Pages/TabsPage.cs [hunk 2]
```diff
@@ -94,7 +94,14 @@ protected override async void OnAppearing()
...
                 {
                     Device.BeginInvokeOnMainThread(async () => await UpdateVaultButtonTitleAsync());
-                    await ForcePasswordResetIfNeededAsync();
+                    try
+                    {
+                        await ForcePasswordResetIfNeededAsync();
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.Value.Exception(ex);
```
Comment (fedemkr) [inline src/App/Pages/TabsPage.cs:103]: @andrebispo5 if an error happens here, should the user be redirected somewhere or is it fine to just ignore the force password reset?

Comment (andrebispo5) [reply src/App/Pages/TabsPage.cs:103]: Like we are doing inside the `ForcePasswordResetIfNeededAsync` method, when there is an error we will only log it.

**Review by fedemkr — DISMISSED** [review]

**Review by fedemkr — APPROVED** [review]

**Pull request merged**